uvtools/dspec.py

```diff
@@ -1235,9 +1241,62 @@ def delay_filter_leastsq(data, flags, sigma, nmax, add_noise=False, freq_units =
     return mdl_array, cn_array, inp_data
 
 
-def sinc_downweight_mat_inv(nchan, df, filter_centers, filter_widths,
-                            filter_factors, cache = {}, wrap = False, wrap_interval=1,
-                            nwraps = 1000, no_regularization = False):
+def delay_interpolation_matrix(nchan, ndelay, wgts, fundamental_period=None, cache={}, taper='none', return_diagnostics=False):
+    '''
+    Computes a foreground interpolation matrix that, when applied to data,
```

Comment:
When writing a docstring, the first line should be a brief imperative statement that summarizes the function's purpose. For example, this could be `Compute a foreground representation matrix for interpolating over flagged channels.`. Then after a blank line, you can have a more extended description as you do here for providing more details to the user. See [this PEP](https://www.python.org/dev/peps/pep-0257/) for more info.

Reply:
I've cleaned this up.